### DISCUSSION TITLE
Renovate still try to use NPM v7, despite i'm using the configuration "force.constraints.node"

### DISCUSSION THREAD

**Discussion by gremo** [renovatebot/renovate#9104] [Request Help] [answered]
I've read the documentation in order to use the "old" NPM v6 ([the constraints setting](https://docs.renovatebot.com/configuration-options/#constraints)). 

My current `renovate.json` is the following:

- Original in my repo: https://github.com/gremo/nest-winston/blob/master/renovate.json
- Changes merged on 03/03/21: https://github.com/gremo/nest-winston/commit/44d2bbb29bd61a2f4ddbb0a58c796ca662db8aef#diff-7b5c8955fc544a11b4b74eddb4115f9cc51c9cf162dbffa60d37eeed82a55a57

```json
{
  "extends": [
    "config:base",
    ":disablePeerDependencies"
  ],
  "packageRules": [
    {
      "depTypeList": [
        "devDependencies"
      ],
      "rangeStrategy": "bump"
    }
  ],
  "force": {
    "constraints": {
      "node": "< 15.0.0"
    }
  }
}
```

I've received two new PRs and they still want to use the new NPM version (look at the diffs in `package-lock.json`:

- https://github.com/gremo/nest-winston/pull/310/files

![npm](https://user-images.githubusercontent.com/1532616/110952925-0f725580-8347-11eb-914a-a8d83b0dfd7f.png)

What I'm doing wrong, why it's not working as documented?

**Comment by rarkins** [comment]
Remove your force statement (recommended), or add an npm constraint to it. Right now your forced constraints are overriding the detected npm constraint, so in the absence of any npm constraint the latest npm is being installed.

**Reply by gremo** [reply]
Sorry I'm pretty new to Renovate. You mean simply:

```
  "constraints": {
    "npm": "< 7"
  }
```

... instead of the `force` key?

**Reply by viceice** [reply]
no, completely remove that from renovate config

**Reply by viceice** [reply]
https://github.com/gremo/nest-winston/pull/312

**Reply by gremo** [reply]
So kind! Thanks!

**Comment by viceice** [comment, marked as answer]
This is not a bug, because you override the detected constraints, renovate has no npm version, so it will install latest npm (as pointed by npmjs).

So best is to not use force override. Alternatively you need to add a npm constraint too.
